### PR TITLE
Add sphinx documentation

### PR DESCRIPTION
**Pull request by pganssle**
Sphinx is the common way of generating documentation for Python, and it doesn't seem to natively support javadocs, so I converted all documentation over to the sphinx-style documentation (`:param x:` instead of `@param x:` and added the `docs` folder.

I also wrote a script that relies on `pypandocs` that converts `README.md` to RST during documentation generation. During setup, it converts the `README.md` to RST and uses it as the long description on PyPI.

### CODE REVIEW

File: cjktools/resources/bilingual_dict.py [hunk 1]
```diff
@@ -61,8 +61,8 @@ def update(self, rhs_entry):
         Update this entry with readings and senses from another homograph.
         The additional readings and senses are added to this entry.
 
-        @param rhs_entry: Another entry to update details from.
-        @type rhs_entry: DictionaryEntry
+        :param DictinoaryEntry rhs_entry:
```
Comment (pganssle) [inline cjktools/resources/bilingual_dict.py]: Oops, typo.

**Comment by pganssle** [comment]
That's weird. It doesn't seem to be picking up `enum34` on the Python2.7 builds anymore, for some reason. I didn't change anything related to that...

**Comment by pganssle** [comment]
Oh, I see what was happening - it was failing because `pypandocs` is not installed and I didn't handle the fallback logic correctly. Fixed now.

**Comment by larsyencken** [comment]
Sorry I'm a bit slow on this. I can run tests on Python 2.7, but for Python 3.5.1 I get a failure to do with the fallbacks:

```
➜  cjktools git:(add_docs) python setup.py test
Traceback (most recent call last):
  File "setup.py", line 23, in <module>
    LONG_DESCRIPTION = convert('README.md', strict=False)
  File "/Users/lars/Dropbox/Documents/personal/cjktools/scripts/load_as_rst.py", line 68, in convert
    text = '\n'.join((text, load_text(fname_in, strict=strict)))
  File "/Users/lars/Dropbox/Documents/personal/cjktools/scripts/load_as_rst.py", line 43, in load_text
    return _fallback_load(fname, strict=strict, parent_exception=e)
NameError: name 'e' is not defined
```

Do you get the same?

**Comment by larsyencken** [comment]
Ok, the `scripts/cibuild.sh` command needs `set -e` inside it, otherwise travis doesn't notice the failure within `setup.py`. If you check the Travis build for Python 3, you'll see it's broken, even though Github gives it a tick.

**Comment by pganssle** [comment]
@larsyencken Good call, didn't realize that could happen. I should probably read up on scoping / binding behavior to better understand why it worked for Python 2.x and not for 3.x. I'm guessing that either the exception doesn't persist outside of the `except` block in 3.x but it does in 2.x or Python 2.x binds earlier.

**Comment by pganssle** [comment]
@larsyencken FYI, when this is done let me know if you want me to set up a readthedocs or pypi document hosting or whatever.

**Comment by larsyencken** [comment]
Thanks for all your work, just merged it now. I'm having a go with readthedocs now, if I get stuck I might ask for some help :)

**Pull request merged**